scrapy/utils/versions.py

```diff
@@ -1,31 +1,46 @@
+from __future__ import annotations
+
 import platform
 import sys
+from importlib.metadata import version
+from warnings import warn
 
-import cryptography
-import cssselect
 import lxml.etree
-import parsel
-import twisted
-import w3lib
 
-import scrapy
+from scrapy.exceptions import ScrapyDeprecationWarning
+from scrapy.settings.default_settings import LOG_VERSIONS
 from scrapy.utils.ssl import get_openssl_version
 
+_DEFAULT_SOFTWARE = ["Scrapy"] + LOG_VERSIONS
+
+
+def _version(item):
+    lowercase_item = item.lower()
+    if lowercase_item == "libxml2":
+        return ".".join(map(str, lxml.etree.LIBXML_VERSION))
+    if lowercase_item == "platform":
+        return platform.platform()
+    if lowercase_item == "pyopenssl":
+        return get_openssl_version()
+    if lowercase_item == "python":
+        return sys.version.replace("\n", "- ")
+    return version(item)
+
+
+def get_versions(
+    software: list | None = None,
+) -> list[tuple[str, str]]:
+    software = software or _DEFAULT_SOFTWARE
+    return [(item, _version(item)) for item in software]
+
 
 def scrapy_components_versions() -> list[tuple[str, str]]:
-    lxml_version = ".".join(map(str, lxml.etree.LXML_VERSION))
-    libxml2_version = ".".join(map(str, lxml.etree.LIBXML_VERSION))
-
-    return [
-        ("Scrapy", scrapy.__version__),
-        ("lxml", lxml_version),
-        ("libxml2", libxml2_version),
-        ("cssselect", cssselect.__version__),
-        ("parsel", parsel.__version__),
-        ("w3lib", w3lib.__version__),
-        ("Twisted", twisted.version.short()),
-        ("Python", sys.version.replace("\n", "- ")),
-        ("pyOpenSSL", get_openssl_version()),
-        ("cryptography", cryptography.__version__),
-        ("Platform", platform.platform()),
-    ]
+    warn(
+        (
+            "scrapy.utils.versions.scrapy_components_versions is deprecated, "
+            "use scrapy.utils.versions.get_versions instead."
+        ),
+        ScrapyDeprecationWarning,
+        stacklevel=2,
+    )
+    return get_versions(_DEFAULT_SOFTWARE)
```

Comment:
Should this call just `get_versions()`, or should the `get_versions()` arg become mandatory? What's your plan for after `scrapy_components_versions()` is removed?